linkup-cli/src/services/mod.rs

```diff
@@ -77,3 +83,64 @@ pub trait BackgroundService<E: std::error::Error> {
             .unwrap();
     }
 }
+
+#[derive(Error, Debug)]
+pub enum PidError {
+    #[error("no pid file: {0}")]
+    NoPidFile(String),
+    #[error("bad pid file: {0}")]
+    BadPidFile(String),
+}
+
+fn get_pid(file_path: &Path) -> Result<Pid, PidError> {
+    if let Err(e) = File::open(file_path) {
+        return Err(PidError::NoPidFile(e.to_string()));
+    }
+
+    match fs::read_to_string(file_path) {
+        Ok(content) => {
+            let pid_u32 = content
+                .trim()
+                .parse::<u32>()
+                .map_err(|e| PidError::BadPidFile(e.to_string()))?;
+
+            Ok(Pid::from_u32(pid_u32))
+        }
+        Err(e) => Err(PidError::BadPidFile(e.to_string())),
+    }
+}
+
+// Get the pid from a pidfile, but only return Some in case the pidfile is valid and the written pid on the file
+// is running.
+pub fn get_running_pid(file_path: &Path) -> Option<Pid> {
+    let pid = match get_pid(file_path) {
+        Ok(pid) => pid,
+        Err(_) => return None,
+    };
+
+    system().process(pid).map(|_| pid)
+}
+
+pub fn stop_pid_file(pid_file: &Path, signal: Signal) -> Result<(), PidError> {
+    match get_running_pid(pid_file) {
+        Some(pid) => match system().process(pid) {
+            Some(process) => {
+                process.kill_with(signal);
+
+                Ok(())
+            }
+            None => Ok(()),
+        },
+        None => Ok(()),
+    }
+}
```

Comment:
Is this one still Result? or just void?

Reply:
It can be just void! Changed on 3426ef071d01c34d496ac8291c643aa673c46e6c